EarlyBird.BusinessLogic/Services/OffersService.cs

```diff
@@ -48,21 +68,48 @@ public ViewOfferDto Add(AddOfferDto offerEntity)
         public bool Delete(int id)
         {
             var offer = offersRepository.GetById(id);
-            if (offer == null) throw new EntityNotFoundException() ;
+            if (offer == null) throw new OfferNotFoundException("Offer could not be found");
 
             return offersRepository.Delete(offer);
         }
 
-        public bool Update(int id, UpdateOfferDto offerEntity)
+        public bool Update(int id, UpdateOfferDto updateOfferEntity)
+        {
+            var offerEntity = offersRepository.GetById(id);
+            if (offerEntity == null)
+                throw new OfferNotFoundException("Offer could not be found");
+
+            offerEntity.Title = updateOfferEntity.Title ?? offerEntity.Title;
+            offerEntity.Cost = updateOfferEntity.Cost == 0 ? offerEntity.Cost : updateOfferEntity.Cost;
+            offerEntity.PublisherId = updateOfferEntity.PublisherId;
+            offerEntity.Description = updateOfferEntity.Description ?? offerEntity.Description;
+            offerEntity.Prerequisites = updateOfferEntity.Prerequisites ?? offerEntity.Prerequisites;
+            offerEntity.Notes = updateOfferEntity.Notes ?? offerEntity.Notes;
+            offerEntity.Status = updateOfferEntity.Status == 0 ? offerEntity.Status : updateOfferEntity.Status;
+            offerEntity.Location = updateOfferEntity.Status == 0 ? offerEntity.Location : mapper.Map<LocationEntity>(updateOfferEntity.Location);
```

Comment:
I think this should be a null check: 
if(updateOfferEntity.Location != null)
    offerEntity.Location =  mapper.Map<LocationEntity>(updateOfferEntity.Location);